core/src/main/scala/org/apache/spark/sql/execution/CompactExecRowToMutableRow.scala

```diff
@@ -221,6 +223,6 @@ trait CompactExecRowToMutableRow extends ResultWasNull {
       }
       i = i + 1
     }
-    mutableRow
+    unsafeproj.apply(mutableRow)
```

Comment:
Thanks for pointing this out. I didn't realize that CachedBatchCreator code is using this too. Fixed.